lib/SIL/GraphFunctionDeviceInfo.cpp

```diff
@@ -14,21 +14,69 @@
 //
 //===----------------------------------------------------------------------===//
 
-
 #include "swift/SIL/GraphFunctionDeviceInfo.h"
 #include "swift/AST/DiagnosticsSIL.h"
 #include "swift/SIL/GraphOperationBuilder.h"
 #include "swift/SIL/GraphOperationInfo.h"
-#include "swift/SIL/SILInstruction.h"
 #include "swift/SIL/SILFunction.h"
+#include "swift/SIL/SILInstruction.h"
 #include "swift/SILOptimizer/Utils/Local.h"
-#include "llvm/ADT/StringRef.h"
+#include "tensorflow/c/c_api_experimental.h"
 #include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
 #include "llvm/Support/CommandLine.h"
+#include <iostream>
+
+#include <unordered_map>
```

Comment:
we usually try and use llvm::DenseMap for its efficiency over unordered_map (the latter involves a lot more memory allocation). 

Some times using DenseMap involves a bit more code (e.g. to set up key data type). PTAL some examples in our TF*.cpp code under SILOptimizer/mandatory/.